EIPS/eip-5170.md

```diff
@@ -0,0 +1,144 @@
+---
+eip: 5170
+title: Asserting authenticity of Client Script for Token Contracts
+description: Provide a method to assert and revoke the authenticity of an ERC5169 client script for token contracts
+author:  Weiwu (@weiwu-zhang), Tore Frederiksen (@jot2re)
+discussions-to: 
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-05-03
+requires: 5169
+---
+
+### Abstract
+
+This ERC describes how to assert the authenticity of the script related to some token or smart contract, regardless of how the script was obtained.
+
+### Motivation
+
+Often NFT authors want to provide some user functionality to their tokens through client scripts. This should be done safely, without opening the user to potential scams. Refer to ERC 5169 examples of such scripts.
+
+Although ERC 5169 specified a way to obtain a set of client scripts through URI, it is inapplicable for token contracts that was issued before the creation of ERC 5169. Furthermore, it lack the finess to address situations such as
+
+- a smart contract might have different scripts for different environments or use-cases. Take a subway token as an example. It might invoke a minimal script at the POS to be sent through NFC (Internet might be slow or inaccessible underground), while advanced functions for user retention, such as rewarding user mascot NFT for continued use, or carbon credit for buying carbon-neutral airfare.
+- in a specific use case, a token's script is often localized, and it doesn't make sense to download and load all language translations.
+- a specific use-case might be compatible with only a specific version of the token's script.
```

Comment:
```suggestion
- A smart contract might have different scripts for different environments or use-cases. Take a subway token as an example. It might invoke a minimal script at the POS to be sent through NFC (Internet might be slow or inaccessible underground), while advanced functions for user retention, such as rewarding user mascot NFT for continued use, or carbon credit for buying carbon-neutral airfare.
- In a specific use case, a token's script is often localized, and it doesn't make sense to download and load all language translations.
- A specific use case might be compatible with only a specific version of the token's script.
```
```suggestion
- a smart contract might have different scripts for different environments or use-cases. Take a subway token as an example. It might invoke a minimal script at the POS to be sent through NFC (Internet might be slow or inaccessible underground), while advanced functions for user retention, such as rewarding user mascot NFT for continued use, or carbon credit for buying carbon-neutral airfare.
- in a specific use case, a token's script is often localized, and it doesn't make sense to download and load all language translations.
- a specific use-case might be compatible with only a specific version of the token's script.
```